string node_name
string name
string value
---